geodata/models/geometry/transform.py

```diff
@@ -0,0 +1,15 @@
+from django.contrib.gis.gdal import SpatialReference
+from django.contrib.gis.geos import GEOSGeometry
+from osgeo import ogr
+from osgeo.osr import CoordinateTransformation, SpatialReference as SpatRef2
+
+from ..constants import DB_SRID
+
+
+def transform_geometry(geometry, source_wkt):
+    """Transform geometry into the database's spatial reference system."""
+    g = ogr.Geometry(wkt=geometry.wkt)
+    source = SpatRef2(source_wkt)
```

Comment:
One of the major issues between Django and GDAL >= 3 is that Django always expects axes to be x-y, but GDAL 3 defaults to the CRS authority order.  Django 3.1 explicitly does `<spatial_reference>.SetAxisMappingStrategy(osgeo.osr.OAMS_TRADITIONAL_GIS_ORDER)`.  We probably need to do that on `source` and `dest`, as, otherwise, the exported geometry may have its coordinates in y-x order.

I have some test cases that should show if this is necessary.

Since this does the geometry transform entirely in GDAL, and only exchanges it with Django via wkt, it looks like it should work generally.

Reply:
> Since this does the geometry transform entirely in GDAL, and only exchanges it with Django via wkt, it looks like it should work generally.

That's what I was hoping. By keeping this transform entirely managed by GDAL, hopefully, this means we can send any geometry to this method and it will output it in the DB spatial reference across version combinations of Django and GDAL

> I have some test cases that should show if this is necessary.

I've been testing this with Landsat imagery (which has the Y,X convention I think) and all seems to be working. It'd be good to get a few other test datasets that we can host somewhere for the tests that we really need to get to in #82

Reply:
> .SetAxisMappingStrategy(osgeo.osr.OAMS_TRADITIONAL_GIS_ORDER)

I had initially tried that yesterday and it yielded the wrong results with my version of Django and GDAL: `3.0.7` and `3.1.0` respectively